### PR TITLE
[pcre] fix build failure on s390x

### LINKED ISSUE #42459: [pcre] Build error on s390x-linux
Package: pcre:s390x-linux@8.45#6

**Host Environment**

- Host: s390x-linux
- Compiler: GNU 10.2.1
-    vcpkg-tool version: 2999-12-31-unknownhash
    vcpkg-scripts version: 7adc2e4d4 2024-10-04 (8 weeks ago)

**To Reproduce**

`vcpkg install `

**Failure logs**

```
-- Downloading https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=cfhcable;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=pilotfiber;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=gigenet;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=versaweb;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=ayera;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=netactuate;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=phoenixnap;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=astuteinternet;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=freefr;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=netcologne;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=deac-riga;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=excellmedia;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=iweb;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=jaist;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=jztkft;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=managedway;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=nchc;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=netix;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=ufpr;https://sourceforge.net/projects/pcre/files/pcre/8.45/pcre-8.45.zip/download?use_mirror=tenet -> pcre-8.45.zip...
CMake Warning (dev) at scripts/cmake/vcpkg_extract_source_archive.cmake:30 (cmake_parse_arguments):
  The REF keyword was followed by an empty string or no value at all.  Policy
  CMP0174 is not set, so cmake_parse_arguments() will unset the arg_REF
  variable rather than setting it to an empty string.
Call Stack (most recent call first):
  scripts/cmake/vcpkg_extract_source_archive_ex.cmake:8 (vcpkg_extract_source_archive)
  scripts/cmake/vcpkg_from_sourceforge.cmake:36 (vcpkg_extract_source_archive_ex)
  buildtrees/versioning_/versions/pcre/50cd82cb1c716b4c31687b01c009a751f4259c8e/portfile.cmake:13 (vcpkg_from_sourceforge)
  scripts/ports.cmake:192 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Extracting source /root/.cache/vcpkg/projects/cppcheck-wheel/downloads/pcre-8.45.zip
-- Applying patch pcre-8.45_suppress_cmake_and_compiler_warnings-errors.patch
-- Applying patch pcre-8.45_fix_postfix_for_debug_Windows_builds.patch
-- Applying patch export-cmake-targets.patch
-- Applying patch pcre-8.4.5_fix_check_function_exists_for_arm-androi_builds.patch
-- Using source at /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean
-- Configuring s390x-linux
-- Building s390x-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /opt/_internal/pipx/venvs/cmake/lib/python3.12/site-packages/cmake/data/bin/cmake --build . --config Debug --target install -- -v -j7
    Working Directory: /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg
    See logs for more information:
      /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/install-s390x-linux-dbg-out.log

Call Stack (most recent call first):
  /cppcheck-wheel/build/py3-none-linux_s390x/vcpkg_installed/s390x-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  /cppcheck-wheel/build/py3-none-linux_s390x/vcpkg_installed/s390x-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  buildtrees/versioning_/versions/pcre/50cd82cb1c716b4c31687b01c009a751f4259c8e/portfile.cmake:48 (vcpkg_cmake_install)
  scripts/ports.cmake:192 (include)



```

<details><summary>/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/install-s390x-linux-dbg-out.log</summary>

```
Change Dir: '/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg'

Run Build Command(s): /tmp/pip-build-env-ptr3ebs9/normal/bin/ninja -v -v -j7 install
[1/75] /opt/rh/devtoolset-10/root/usr/bin/gcc  -pthread -DHAVE_CONFIG_H -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean -fPIC -g -MD -MT CMakeFiles/pcre.dir/pcre_chartables.c.o -MF CMakeFiles/pcre.dir/pcre_chartables.c.o.d -o CMakeFiles/pcre.dir/pcre_chartables.c.o -c /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg/pcre_chartables.c
[2/75] /opt/rh/devtoolset-10/root/usr/bin/gcc  -pthread -DHAVE_CONFIG_H -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean -fPIC -g -MD -MT CMakeFiles/pcre.dir/pcre_config.c.o -MF CMakeFiles/pcre.dir/pcre_config.c.o.d -o CMakeFiles/pcre.dir/pcre_config.c.o -c /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_config.c
[3/75] /opt/rh/devtoolset-10/root/usr/bin/gcc  -pthread -DHAVE_CONFIG_H -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean -fPIC -g -MD -MT CMakeFiles/pcre.dir/pcre_byte_order.c.o -MF CMakeFiles/pcre.dir/pcre_byte_order.c.o.d -o CMakeFiles/pcre.dir/pcre_byte_order.c.o -c /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_byte_order.c
[4/75] /opt/rh/devtoolset-10/root/usr/bin/gcc  -pthread -DHAVE_CONFIG_H -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean -fPIC -g -MD -MT CMakeFiles/pcre.dir/pcre_fullinfo.c.o -MF CMakeFiles/pcre.dir/pcre_fullinfo.c.o.d -o CMakeFiles/pcre.dir/pcre_fullinfo.c.o -c /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_fullinfo.c
[5/75] /opt/rh/devtoolset-10/root/usr/bin/gcc  -pthread -DHAVE_CONFIG_H -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean -fPIC -g -MD -MT CMakeFiles/pcre.dir/pcre_globals.c.o -MF CMakeFiles/pcre.dir/pcre_globals.c.o.d -o CMakeFiles/pcre.dir/pcre_globals.c.o -c /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_globals.c
[6/75] /opt/rh/devtoolset-10/root/usr/bin/gcc  -pthread -DHAVE_CONFIG_H -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean -fPIC -g -MD -MT CMakeFiles/pcre.dir/pcre_get.c.o -MF CMakeFiles/pcre.dir/pcre_get.c.o.d -o CMakeFiles/pcre.dir/pcre_get.c.o -c /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_get.c
[7/75] /opt/rh/devtoolset-10/root/usr/bin/gcc  -pthread -DHAVE_CONFIG_H -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean -fPIC -g -MD -MT CMakeFiles/pcre.dir/pcre_maketables.c.o -MF CMakeFiles/pcre.dir/pcre_maketables.c.o.d -o CMakeFiles/pcre.dir/pcre_maketables.c.o -c /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_maketables.c
[8/75] /opt/rh/devtoolset-10/root/usr/bin/gcc  -pthread -DHAVE_CONFIG_H -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean -fPIC -g -MD -MT CMakeFiles/pcre.dir/pcre_newline.c.o -MF CMakeFiles/pcre.dir/pcre_newline.c.o.d -o CMakeFiles/pcre.dir/pcre_newline.c.o -c /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_newline.c
[9/75] /opt/rh/devtoolset-10/root/usr/bin/gcc  -pthread -DHAVE_CONFIG_H -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean -fPIC -g -MD -MT CMakeFiles/pcre.dir/pcre_ord2utf8.c.o -MF CMakeFiles/pcre.dir/pcre_ord2utf8.c.o.d -o CMakeFiles/pcre.dir/pcre_ord2utf8.c.o -c /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_ord2utf8.c
[10/75] /opt/rh/devtoolset-10/root/usr/bin/gcc  -pthread -DHAVE_CONFIG_H -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean -fPIC -g -MD -MT CMakeFiles/pcre.dir/pcre_jit_compile.c.o -MF CMakeFiles/pcre.dir/pcre_jit_compile.c.o.d -o CMakeFiles/pcre.dir/pcre_jit_compile.c.o -c /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_jit_compile.c
FAILED: CMakeFiles/pcre.dir/pcre_jit_compile.c.o 
/opt/rh/devtoolset-10/root/usr/bin/gcc  -pthread -DHAVE_CONFIG_H -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean -fPIC -g -MD -MT CMakeFiles/pcre.dir/pcre_jit_compile.c.o -MF CMakeFiles/pcre.dir/pcre_jit_compile.c.o.d -o CMakeFiles/pcre.dir/pcre_jit_compile.c.o -c /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_jit_compile.c
/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_jit_compile.c:65:2: error: #error Unsupported architecture
   65 | #error Unsupported architecture
      |  ^~~~~
/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_jit_compile.c: In function ‘pcre_jit_free_unused_memory’:
/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_jit_compile.c:11844:1: warning: implicit declaration of function ‘sljit_free_unused_memory_exec’; did you mean ‘pcre_jit_free_unused_memory’? [-Wimplicit-function-declaration]
11844 | sljit_free_unused_memory_exec();
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      | pcre_jit_free_unused_memory
[11/75] /opt/rh/devtoolset-10/root/usr/bin/gcc  -pthread -DHAVE_CONFIG_H -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean -fPIC -g -MD -MT CMakeFiles/pcre.dir/pcre_dfa_exec.c.o -MF CMakeFiles/pcre.dir/pcre_dfa_exec.c.o.d -o CMakeFiles/pcre.dir/pcre_dfa_exec.c.o -c /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_dfa_exec.c
[12/75] /opt/rh/devtoolset-10/root/usr/bin/gcc  -pthread -DHAVE_CONFIG_H -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean -fPIC -g -MD -MT CMakeFiles/pcre.dir/pcre_refcount.c.o -MF CMakeFiles/pcre.dir/pcre_refcount.c.o.d -o CMakeFiles/pcre.dir/pcre_refcount.c.o -c /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_refcount.c
[13/75] /opt/rh/devtoolset-10/root/usr/bin/gcc  -pthread -DHAVE_CONFIG_H -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean -fPIC -g -MD -MT CMakeFiles/pcre.dir/pcre_string_utils.c.o -MF CMakeFiles/pcre.dir/pcre_string_utils.c.o.d -o CMakeFiles/pcre.dir/pcre_string_utils.c.o -c /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_string_utils.c
[14/75] /opt/rh/devtoolset-10/root/usr/bin/gcc  -pthread -DHAVE_CONFIG_H -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean -fPIC -g -MD -MT CMakeFiles/pcre.dir/pcre_study.c.o -MF CMakeFiles/pcre.dir/pcre_study.c.o.d -o CMakeFiles/pcre.dir/pcre_study.c.o -c /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_study.c
[15/75] /opt/rh/devtoolset-10/root/usr/bin/gcc  -pthread -DHAVE_CONFIG_H -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean -fPIC -g -MD -MT CMakeFiles/pcre.dir/pcre_compile.c.o -MF CMakeFiles/pcre.dir/pcre_compile.c.o.d -o CMakeFiles/pcre.dir/pcre_compile.c.o -c /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_compile.c
[16/75] /opt/rh/devtoolset-10/root/usr/bin/gcc  -pthread -DHAVE_CONFIG_H -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/s390x-linux-dbg -I/root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean -fPIC -g -MD -MT CMakeFiles/pcre.dir/pcre_exec.c.o -MF CMakeFiles/pcre.dir/pcre_exec.c.o.d -o CMakeFiles/pcre.dir/pcre_exec.c.o -c /root/.cache/vcpkg/projects/cppcheck-wheel/buildtrees/pcre/src/pcre-8-04f0ec285a.clean/pcre_exec.c
ninja: build stopped: subcommand failed.
```
</details>

**Additional context**

<details><summary>vcpkg.json</summary>

```
{
  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/main/docs/vcpkg.schema.json",
  "name": "cppcheck-wheel",
  "description": "Package cppcheck as a python wheel.",
  "builtin-baseline": "7adc2e4d49e8d0efc07a369079faa6bc3dbb90f3",
  "homepage": "https://github.com/msclock/cppcheck-wheel",
  "dependencies": [
    {
      "name": "cppcheck",
      "features": [
        "have-rules"
      ]
    },
    "cmake-modules"
  ],
  "overrides": [
    {
      "name": "cppcheck",
      "version": "2.16.0"
    },
    {
      "name": "cmake-modules",
      "version": "1.6.8"
    }
  ],
  "vcpkg-configuration": {
    "registries": [
      {
        "kind": "git",
        "baseline": "60a98c312d8be8aa0013a69bb5158788702d8700",
        "repository": "https://github.com/msclock/cmake-registry",
        "packages": [
          "cmake-modules",
          "robotology-cmake-ycm",
          "vcpkg-cmake-module"
        ]
      }
    ],
    "overlay-ports": [
      "cmake/vcpkg/ports"
    ],
    "overlay-triplets": [
      "cmake/vcpkg/triplets"
    ]
  }
}

```
</details>

## Reproduce

Prepare s390 env
```bash
docker run --rm --privileged docker.io/tonistiigi/binfmt:latest --install all
docker run --rm -it quay.io/pypa/manylinux2014_s390x:2024.11.16-1 bash
```
Reproduce bugs
```bash
git clone https://github.com/msclock/cppcheck-wheel.git
cd cppcheck-wheel/
git fetch origin f6e6d1050a158d76cd72d51d5df977525418dd31
git checkout f6e6d1050a158d76cd72d51d5df977525418dd31
/opt/python/cp310-cp310/bin/pip install virtualenv
/opt/python/cp310-cp310/bin/virtualenv .venv
source .venv/bin/activate
export VCPKG_FORCE_SYSTEM_BINARIES=1
yum install zip -y
pip install . -v
```

### PR DESCRIPTION
**Pull request by msclock**
Fixes #42459

### CODE REVIEW

File: ports/pcre/portfile.cmake [hunk 1]
```diff
@@ -20,7 +20,7 @@ vcpkg_from_sourceforge(
 )
 
 set(IS_PCRE_SUPPORT_JIT YES)
-if(VCPKG_TARGET_ARCHITECTURE MATCHES "loongarch")
+if(VCPKG_TARGET_ARCHITECTURE MATCHES "(loongarch|^s390|^arm)")
```
Comment (dg0yt) [inline ports/pcre/portfile.cmake:23]: I don't see failures for arm android/osx/windows CI.
```suggestion
if(VCPKG_TARGET_ARCHITECTURE MATCHES "(loongarch|^s390)")
```

File: versions/baseline.json [hunk 2]
```diff
@@ -6274,7 +6274,7 @@
     },
     "pcre": {
       "baseline": "8.45",
-      "port-version": 5
+      "port-version": 6
```
Comment (dg0yt) [inline versions/baseline.json:6277]: You must update and commit `port/pcre/vcpkg.json` first.
`versions` is managed by `vcpkg x-add-version [--overwrite-version] pcre`.

File: ports/pcre/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/pcre/portfile.cmake:23]: (Maybe CI doesn't test jit. IDK.)

Comment (msclock) [reply ports/pcre/portfile.cmake:23]: Ah, I succeded to build it on armv7l but missed it before.

**Comment by msclock** [comment]
It is werid and still has conflicts.

**Comment by dg0yt** [comment]
It is not so weird: You started with a commit from August 2023(!). Rebase on HEAD of master.

**Comment by msclock** [comment]
> You started with a commit from August 2023(!)

@dg0yt Thanks for pointing out it. Rebased now.

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**